src/migmose/parsing.py

```diff
@@ -0,0 +1,85 @@
+"""
+contains functions for file handling and parsing.
+"""
+
+import json
+from pathlib import Path
+from typing import Generator, Union
+
+import click
+import docx  # type: ignore[import]
+from docx.document import Document  # type: ignore[import]
+from docx.oxml import CT_Tbl  # type: ignore[import]
+from docx.table import Table, _Cell  # type: ignore[import]
+from docx.text.paragraph import Paragraph  # type: ignore[import]
+from loguru import logger
+from maus.edifact import EdifactFormat
+
+
+def find_file_to_type(message_types: list[EdifactFormat], input_dir: Path) -> dict[EdifactFormat, Path]:
+    """
+    finds the file with the message type in the input directory
+    """
+    file_dict = {}
+    for message_type in message_types:
+        list_of_all_files = [
+            file for file in input_dir.iterdir() if message_type in file.name and file.suffix == ".docx"
+        ]
+        if len(list_of_all_files) == 1:
+            file_dict[message_type] = list_of_all_files[0]
+        elif len(list_of_all_files) > 1:
+            logger.warning(f"⚠️ There are several files for {message_type}.", fg="red")
+        else:
+            logger.warning(f"⚠️ No file found for {message_type}.", fg="red")
+    if file_dict:
+        return file_dict
+    logger.error("⚠️ No files found in the input directory.", fg="red")
+    raise click.Abort()
+
+
+def preliminary_output_as_json(table: list[str], message_type: EdifactFormat, output_dir: Path) -> None:
+    """
+    writes the preliminary output as json
+    """
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
das kannst du vereinfachen

```suggestion
output_dir.mkdir(parents=True, exist_ok=True)
```
das `exist_ok` hilft dir genau dabei.

> will not raise an exception if the directory already exists (exist_ok=True)

Reply:
62978f072692b1564cb5494f814ae15a8748dbd3